langfuse-core/src/prompts/promptClients.ts

```diff
@@ -0,0 +1,96 @@
+import mustache from "mustache";
+
+import type { ChatMessage, ChatPrompt, CreateLangfusePromptResponse, TextPrompt } from "../types";
+
+abstract class BasePromptClient {
+  public readonly name: string;
+  public readonly version: number;
+  public readonly config: unknown;
+
+  constructor(prompt: CreateLangfusePromptResponse) {
+    this.name = prompt.name;
+    this.version = prompt.version;
+    this.config = prompt.config;
+  }
+
+  abstract compile(variables?: Record<string, string>): string | ChatMessage[];
+
+  public abstract getLangchainPrompt(): string | ChatMessage[];
+
+  protected _transformToLangchainVariables(content: string): string {
+    return content.replace(/\{\{(.*?)\}\}/g, "{$1}");
```

Comment:
I would dismiss this:
- Users of our SDK need to create correct prompts, otherwise they will slow down their own application which is using our SDK
- Our users need to ensure that their users do not insert malicious code into the prompts.

Reply:
WDYT, @hassiebp